uber/models.py

```diff
@@ -1300,6 +1346,19 @@ def _staffing_adjustments(self):
     def _email_adjustment(self):
         self.email = normalize_email(self.email)
 
+    @presave_adjustment
+    def _find_promo_code(self):
+        if self.promo_code:
+
+
+    @presave_adjustment
+    def _use_promo_code(self):
```

Comment:
Even though we have it in model_checks, we should probably ALSO check here to make sure a code can be used before applying its price to an attendee.

This also reveals a potential tricky decision -- promo codes overload `overridden_price.` I think I'm okay with this -- we could theoretically add promo codes to the logic of `badge_cost` but then how would we check to make sure the attendee's real badge cost isn't cheaper than the promo code?